karld/run_together.py

```diff
@@ -133,6 +137,95 @@ def pool_run_files_to_files(file_to_file, in_dir, filter_func=None):
         return list(pool.map(file_to_file, results_final))
 
 
+def distribute_run_to_runners(items_func, in_url, reader=None, batch_size=1100):
+    """
+    With a multi-process pool, map batches of items from
+    file to an items processing function.
+
+    The reader callable should be as fast as possible to
+    reduce data feeder cpu usage. It should do the minimal
+    to produce discrete units of data, save any decoding
+    for the items function.
+
+    :param items_func: Callable that takes multiple items of the data.
+    :param reader: URL reader callable.
+    :param in_url: Url of content
+    :param batch_size: size of batches.
+    """
+    from concurrent.futures import ProcessPoolExecutor
+
+    if not reader:
+        reader = i_read_buffered_binary_file
+
+    stream = reader(in_url)
+    batches = i_batch(batch_size, stream)
+
+    with ProcessPoolExecutor() as pool:
+        return list(pool.map(items_func, batches))
+
+
+def distribute_multi_run_to_runners(items_func, in_dir,
+                                    reader=None,
+                                    batch_size=1100,
+                                    filter_func=None):
+    """
+    With a multi-process pool, map batches of items from
+    file to an items processing function.
+
+    The reader callable should be as fast as possible to
+    reduce data feeder cpu usage. It should do the minimal
+    to produce discrete units of data, save any decoding
+    for the items function.
+
+    :param items_func: Callable that takes multiple items of the data.
+    :param reader: URL reader callable.
+    :param in_url: Url of content
+    :param batch_size: size of batches.
+    :param filter_func: a function that returns True for desired paths names.
+    """
+    from concurrent.futures import ProcessPoolExecutor
+    from multiprocessing import cpu_count
+
+    if not reader:
+        reader = i_read_buffered_binary_file
+
+    paths_names = i_walk_dir_for_filepaths_names(in_dir)
```

Comment:
Unlike distribute_run_to_runners, this is for only reading files. It could be more source agnostic if the walker function was injectable. Maybe this should just take a iterator of urls for the reader to read from.